[TRIVIAL] Update Linux Build Instructions

crypticrabbit opened the pull request:
## High Level Overview of Change
The rippled Linux build instructions have conflicting information - in some places they suggest repo versions of protobuf are sufficient while other lines explicitly mandate that v2 is required. Debian 10+ as well as Ubuntu 18.04+ ship with protobuf v3.

I also removed a reference to Ubuntu 16.04, since it seems like many of the dependence (including protobuf and the compiler) would be out of date, so it would be far easier to build on 18.04. However, it might make sense to leave the reference in, since 16.04 is not EoL.

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates
- [ ] Release

legleux commented on Builds/linux/README.md:
```diff
@@ -6,10 +6,8 @@ builds, including docker based builds for those distributions, please consult
 the [rippled-package-builder](https://github.com/ripple/rippled-package-builder)
```
This second line is irrelevant. This functionality has been subsumed by rippled the repo. (and I have a pull request the removes it again 🙄 )

legleux replied:
"number of processors" lines 41,152,173,183, referenced several times could be:
`$(echo $(nproc)-2 | bc)`
line 103 says `my_build` which won't be `.gitignore`d
I guess I can't comment on stuff you haven't changed.

crypticrabbit replied:
I would be glad to make those changes (and I might even have time to do it on a cleanly forked repo, so crufty PRs don't make their way in).

However, it will be Thursday at the earliest, since I'm working tonight.

crypticrabbit commented:
Thanks for the additional edits, @legleux. I've updated the build instructions, so they now automatically detect the number of processor cores. I also changed the `my_build` directory to `build`, so it is picked up by `.gitignore`.

nbougalis commented:
Hey @crypticrabbit, can you please squash this commit down to one or two, as appropriate, and rebase it to the latest **`develop`**? Thanks.

intelliot commented:
(notes: will squash this to 1 commit)
